test/integration/tests/dependencies.ts

```diff
@@ -37,6 +47,34 @@ describe('e2e-dependencies', function () {
       }
     });
 
+    it('contract category correct', async function () {
+      for (let i = 0; i < proposalNames.length; i++) {
+        const proposalName = proposalNames[i];
+        const contracts = proposals[proposalName].affectedContractSignoff;
+        const deprecated = proposals[proposalName].deprecatedContractSignoff;
+
+        doLogging && console.log(`Checking proposal: ${proposalName}`);
+        doLogging && console.log(`Proposal affects contracts: ${contracts}`);
+
+        for (let j = 0; j < contracts.length; j++) {
+          const contract = contracts[j];
+          const category = addresses[contract].category;
+          expect(category).to.not.be.equal('External');
+          expect(category).to.not.be.equal('Deprecated');
+
+          expect(deprecated).to.not.contain(contract);
+        }
+
+        for (let j = 0; j < deprecated.length; j++) {
+          const contract = deprecated[j];
+          const category = addresses[contract].category;
+          expect(category).to.be.equal('Deprecated');
+
+          expect(contracts).to.not.contain(contract);
+        }
+      }
+    });
+
     it('all dependencies bidirectional', async function () {
```

Comment:
I know this wasn't an edit on this pr but the string "Check dependencies all dependencies bidirectional" doesn't tell me what it's checking, maybe something like "Check dependencies are listed bidirectionally" would?

Reply:
changed this and others